dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/context/ProgramInstanceSupplier.java

```diff
@@ -80,26 +81,25 @@ public Map<String, ProgramInstance> get( ImportOptions importOptions,
 
         // Collect all the program instance UIDs to pass as SQL query argument
         Set<String> programInstanceUids = events.stream()
-            .filter( e -> StringUtils.isNotEmpty( e.getEnrollment() ) )
-            .map( Event::getEnrollment ).collect( Collectors.toSet() );
-
-        Map<String, ProgramInstance> programInstances = new HashMap<>();
-
-        if ( !programInstanceUids.isEmpty() )
-        {
-            // Create a bi-directional map enrollment uid -> event uid
-            Multimap<String, String> programInstanceToEvent = HashMultimap.create();
-            for ( Event event : events )
-            {
-                programInstanceToEvent.put( event.getEnrollment(), event.getUid() );
-            }
+            .map( Event::getEnrollment )
+            .filter( StringUtils::isNotEmpty ).collect( Collectors.toSet() );
+
+        Map<String, ProgramInstance> programInstances = Optional.of( programInstanceUids )
```

Comment:
Is `programInstanceUids` ever null? Should it not rather be an empty set if there are no events with a non empty enrollment 🤔